### DISCUSSION TITLE
Allow datasource TLS ca/cert/key file to be set from a file path instead of explicit certificates

### DISCUSSION THREAD

**Discussion by j2udev** [grafana/grafana#52265] [Data Sources]
Currently, Grafana datasources can't configure TLS settings from a filepath.  This is frustrating because it requires manual intervention. When using a service mesh (like Istio) often the certificate is rotated for us and by volumemounting the certificate in the appropriate locations, we can give Grafana access to the TLS certs within its container.  If the datasource TLS configuration could be configured based on filepaths, we would never need to touch the data source config after initially setting it up.  Also it would be generally more secure for configuration as code scenarios.

From the docs, https://grafana.com/docs/grafana/latest/administration/provisioning/, we see these 3 fields:

```
tlsCACert | string | HTTP*, MySQL, PostgreSQL | CA cert for out going requests
tlsClientCert | string | HTTP*, MySQL, PostgreSQL | TLS Client cert for outgoing requests
tlsClientKey | string | HTTP*, MySQL, PostgreSQL | TLS Client key for outgoing requests
```
I would expect there could be 3 new fields that would be mutually exclusive to the above... perhaps could just add a `File`/`Path`/`FilePath` suffix to them:
```
tlsCACertPath | string | HTTP*, MySQL, PostgreSQL | Path to the CA cert for out going requests
tlsClientCertPath | string | HTTP*, MySQL, PostgreSQL | Path to the TLS Client cert for outgoing requests
tlsClientKeyPath | string | HTTP*, MySQL, PostgreSQL | Path to the TLS Client key for outgoing requests
```

**Comment by daynebenn** [comment]
https://github.com/grafana/grafana/discussions/44296#discussioncomment-2515929

Using the grafana variable expansion file provider `$__file{/path/to/file}` works. Although it's not in the provisioning docs it is explained in the [Configure Grafana docs](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#variable-expansion)

**Comment by gabor** [comment]
hi @j2udev , the behavior regarding this may be different from datasource to datasource. which datasource are you trying to use with the tls certificates? (mysql ? postgres? something else?) thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
